### DISCUSSION TITLE
Combine multiple queries into single SELECT query

### DISCUSSION THREAD

**Discussion by tqwewe** [diesel-rs/diesel#2645] [Ideas]
### Versions

- **Rust:** 1.23.1
- **Diesel:** 1.4.4
- **Database:** Postgres
- **Operating System** MacOS

### Feature Flags

- **diesel:** `["postgres", "r2d2"]`

## Problem Description

Given the following SQL query:
```sql
SELECT * FROM
        /* Has previous page */
        (SELECT COUNT(*) > 0 as has_previous_page
            FROM reviews
            WHERE store_id = $1 AND id < $2) as has_previous_page,

        /* Has next page */
        (SELECT COUNT(*) > 0 as has_next_page
            FROM reviews
            WHERE store_id = $1 AND id > ($3 + 1)) as has_next_page,

        /* Total count */
        (SELECT COUNT(*) as total_count
            FROM reviews
            WHERE store_id = $1) as total_count
```

How can I execute this in Diesel?

So far I have:
```rust
let has_previous_page_query = reviews
        .select(count_star().gt(0))
        .filter(store_id.eq(store_id_param))
        .filter(id.lt(after))
        .single_value();

let has_next_page_query = reviews
        .select(count_star().gt(0))
        .filter(store_id.eq(store_id_param))
        .filter(id.gt(after + first))
        .single_value();

let total_count_query = reviews
        .count()
        .filter(store_id.eq(store_id_param))
        .single_value();

// Need to combine the three queries with `SELECT * FROM query1, query2, query3` in diesel
```

I am not sure how I can run a select query on all three of these queries in a single query. Is this functionality missing from Diesel? Or is it something I am missing?

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues)  and the [disussion forum](https://github.com/diesel-rs/diesel/discussions)  for similar possible closed issues.
 
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)
- [x] This issue can be reproduced without requiring a third party crate

**Comment by weiznich** [comment]
Converted to a discussion post, as this is currently not supported by diesel and therefore a feature request. Feel free to fill in the missing points from [contributing guide](https://github.com/diesel-rs/diesel/blob/master/CONTRIBUTING.md#submitting-feature-requests)
